src/nn/webgl/WebGLModel.js

```diff
@@ -278,24 +279,16 @@ export default class WebGLModel {
     });
   }
 
-  async _executeGlSubgraph(subgraph) {
-    tf.webgl.forceHalfFloat();
-    console.info('WEBGL_FORCE_F16_TEXTURES : ',tf.ENV.getBool('WEBGL_FORCE_F16_TEXTURES'));
-    console.info('floatPercision : ',tf.backend().floatPrecision());    
+  async _executeGlSubgraph(subgraph) {   
     for (const operation of subgraph.operations) {
       tf.tidy(() => this._executeGlOperation(operation));
     }
 
-    // fence
-    // const queue = [];
     for (const tensorId of subgraph.outputs) {
       const buffer = this._nnOperands[tensorId];
       const operand = this._operands[tensorId];
-      buffer.set(operand.dataSync());
```

Comment:
What's the reason that not using `dataSync`? Is it because it is deprecated?

Reply:
Because tfjs-backend-webgpu does not support datasync at the moment. Some operands are storaged in `this._model._operands`，we upload them to the WebGL texture and then use `dataSync` to read them back. It is not necessary

Reply:
Hi all,
I'm a little worried about its performance. Copy-pasted from my [earlier comment](https://github.com/intel/webml-polyfill/issues/115#issuecomment-485352957):
> `data()` uses [fence polling](https://github.com/tensorflow/tfjs/blob/1fa1beb3b08e00065b830bc2dcec435bbf5a8af3/tfjs-core/src/backends/webgl/backend_webgl.ts#L404) to sync while `dataSync` uses blocking call [`readPixels`](https://github.com/tensorflow/tfjs/blob/1fa1beb3b08e00065b830bc2dcec435bbf5a8af3/tfjs-core/src/backends/webgl/gpgpu_util.ts#L291) to sync. From my comparisons of these two approaches, `data()` could be much slower.

Also, when a subgraph has multiple outputs (in hybrid mode), waiting for outputs **serially** might not be a good idea.

Reply:
Thanka, I will not call tf.data () again, wait till tfjs-backend-webgpu's dataSync () function is released.